zerver/tests/test_message_fetch.py

```diff
@@ -1237,6 +1321,30 @@ def test_ok_to_include_history(self) -> None:
         ]
         self.assertTrue(ok_to_include_history(narrow, user_profile, False))
 
+        narrow = [
+            NarrowParameter(operator="channels", operand="archived"),
+            NarrowParameter(operator="is", operand="mentioned"),
+        ]
+        self.assertFalse(ok_to_include_history(narrow, user_profile, False))
+
+        narrow = [
+            NarrowParameter(operator="channels", operand="archived"),
+            NarrowParameter(operator="is", operand="unread"),
+        ]
+        self.assertFalse(ok_to_include_history(narrow, user_profile, False))
+
+        narrow = [
+            NarrowParameter(operator="channels", operand="archived"),
+            NarrowParameter(operator="is", operand="alerted"),
+        ]
+        self.assertFalse(ok_to_include_history(narrow, user_profile, False))
+
+        narrow = [
+            NarrowParameter(operator="channels", operand="archived"),
+            NarrowParameter(operator="is", operand="resolved"),
+        ]
+        self.assertFalse(ok_to_include_history(narrow, user_profile, False))
```

Comment:
I think this should probably test that if you had `channels:public` too, `ok_to_include_history` becomes true.

(The question is mainly about whether any channels with protected history can exist).